File: week2/first-unique-character-in-a-string.java
// Uses the multiset class in Multiset.java
class Solution {
    public int firstUniqChar(String s) {
        Multiset<Character> ms = new Multiset<>();
        for (int i = 0; i < s.length(); i++) {
            ms.add(s.charAt(i));
        }
        for (int i = 0; i < s.length(); i++) {
            if (ms.count(s.charAt(i)) == 1) {
                return i;
            } 
        }
        return -1;
    }
}

File: summer2021/unit2/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        counts = {}
        for c in s:
            if c in counts:
                counts[c] += 1
            else:
                counts[c] = 1
        countsList = []
        for c in s:
            countsList.append(counts[c])
        for i in range(len(countsList)):
            if countsList[i] == 1:
                return i
        return -1